### PR TITLE
Delete Operation

### PR DESCRIPTION
**Pull request by SSTechInd-Tec**
This is the delete operation PR

### CODE REVIEW

File: app/api/add/router.tsx
Comment (hakima3) [inline app/api/add/router.tsx:1]: router extension should be .ts not .tsx, you are not returning jsx

Comment (SSTechInd-Tec) [reply app/api/add/router.tsx:1]: okey

File: app/todos/list/[author]/[id]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    const data = await fetchData()
+
+    let matched_todo = data.find(item => item.author === params.author && item.id === parseInt(params.id))
+
+
+    return (
+        <div>
+            {
+                <div role="alert" className="alert alert-success my-2">
+                    {matched_todo.done ? <svg xmlns="http://www.w3.org/2000/svg" className="stroke-current shrink-0 h-6 w-6" fill="none" viewBox="0 0 24 24"><path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M9 12l2 2 4-4m6 2a9 9 0 11-18 0 9 9 0 0118 0z" /></svg>:<svg xmlns="http://www.w3.org/2000/svg" className="stroke-current shrink-0 h-6 w-6" fill="none" viewBox="0 0 24 24"><path strokeLinecap="round" strokeLinejoin="round" strokeWidth="2" d="M10 14l2-2m0 0l2-2m-2 2l-2-2m2 2l2 2m7-2a9 9 0 11-18 0 9 9 0 0118 0z" /></svg>}
```
Comment (skypencil) [inline app/todos/list/[author]/[id]/page.tsx:22]: add the `svg` as an external file. It should not be inline.

File: app/api/add/router.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    const body = await request.json()
+
+    const new_todo = {
+        id: data.length + 1,
+        todo: body.todo,
+        author: body.author,
+        done: false
+    }
+
+    const res = await fetch("http://localhost:3000/api/", {
```
Comment (skypencil) [inline app/api/add/router.tsx:14]: please add a `tag` to this fetch and have an example of cache re validation as well. Maybe re-validate the cache every time a new item is added.

**Review by skypencil — COMMENTED** [review]
I couldn't think of anything else. I have some minor comments please address those. also make sure you have an example of `cache` re validation by `tag`.

If any of my comments are not clear to you please talk to me about it on slack. Also I saw Hakima's comments. Please address those as well.

**Pull request merged**